### PR TITLE
feat: Supports Windows agent pools disable OutBoundNAT

### PR DESCRIPTION
**Pull request by ShiqianTao**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Support Windows agent pools disable OutBoundNAT
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/agent/datamodel/types.go [hunk 1]
```diff
@@ -662,7 +662,8 @@ type AgentPoolProfile struct {
 	MessageOfTheDay       string               `json:"messageOfTheDay,omitempty"`
 	// This is a new property and all old agent pools do no have this field. We need to keep the default
 	// behavior to reboot Windows node when it is nil
-	NotRebootWindowsNode *bool `json:"notRebootWindowsNode,omitempty"`
+	NotRebootWindowsNode *bool                    `json:"notRebootWindowsNode,omitempty"`
+	WindowsProfile       *AgentPoolWindowsProfile `json:"agentPoolWindowsProfile,omitempty"`
```
Comment (AbelHu) [inline pkg/agent/datamodel/types.go:666]: use `agentPoolWindowsProfile` to differentiate from the property in cluster level?

File: staging/cse/windows/azurecnifunc.ps1 [hunk 2]
```diff
@@ -93,6 +93,30 @@ function Set-AzureCNIConfig
...
+            Name = 'EndpointPolicy'
+            Value = $valueObj
+        }
+
+        # AdditionalArgs[0] is OutBoundNAT with ExceptionList; AdditionalArgs[1] is Route.
+        # Disable OutBoundNAT&Route if WinDSR is enabled. Otherwise, only disable OutBoundNAT and reserve Route.
+        if ($global:KubeproxyFeatureGates.Contains("WinDSR=true")) {
+            $configJson.plugins[0].AdditionalArgs = @($jsonContent)
+        } else {
+            $configJson.plugins[0].AdditionalArgs[0] = $jsonContent
```
Comment (AbelHu) [inline staging/cse/windows/azurecnifunc.ps1:116]: It will remove other settings, for example, settings when IsDualStackEnabled is true

Comment (AbelHu) [reply staging/cse/windows/azurecnifunc.ps1:116]: From the line 63-74, AdditionalArgs is always an array.
```
    $osBuildNumber = (get-wmiobject win32_operatingsystem).BuildNumber
    if ($osBuildNumber -le 17763){
        # In WS2019 and below rules in the exception list are generated by dropping the prefix lenght and removing duplicate rules.
        # If multiple execptions are specified with different ranges we should only include the broadest range for each address.
        # This issue has been addressed in 19h1+ builds

        $processedExceptions = GetBroadestRangesForEachAddress $exceptionAddresses
        Write-Log "Filtering CNI config exception list values to work around WS2019 issue processing rules. Original exception list: $exceptionAddresses, processed exception list: $processedExceptions"
        $configJson.plugins.AdditionalArgs[0].Value.ExceptionList = $processedExceptions
    }
    else {
        $configJson.plugins.AdditionalArgs[0].Value.ExceptionList = $exceptionAddresses
    }
```

File: pkg/agent/datamodel/types.go (hunk 1 above)
Comment (AbelHu) [reply pkg/agent/datamodel/types.go:666]: I think that we can also use agentPoolWindowsProfile in API. If not, we can use `agentPoolWindowsProfile` in code but expose windowsProfile in API

Comment (ShiqianTao) [reply pkg/agent/datamodel/types.go]: Updated to AgentPoolWindowsProfile.

File: staging/cse/windows/azurecnifunc.ps1 (hunk 2 above)
Comment (ShiqianTao) [reply staging/cse/windows/azurecnifunc.ps1]: Split this logic from this PR.

File: parts/windows/kuberneteswindowssetup.ps1 [hunk 3]
```diff
@@ -184,6 +184,9 @@ $global:TLSBootstrapToken = "{{GetTLSBootstrapTokenForKubeConfig}}"
 
 $global:IsNotRebootWindowsNode = [System.Convert]::ToBoolean("{{GetVariable "isNotRebootWindowsNode" }}");
 
+# Disable OutBoundNAT in Azure CNI configuration
+$global:IsDisableWindowsOutboundNat = [System.Convert]::ToBoolean("{{GetVariable "isDisableWindowsOutboundNat" }}");
```
Comment (AbelHu) [inline parts/windows/kuberneteswindowssetup.ps1:188]: Should we set `$WindowsCSEScriptsPackage` to "aks-windows-cse-scripts-v0.0.14.zip" in this PR?

Comment (ShiqianTao) [reply parts/windows/kuberneteswindowssetup.ps1:188]: Updated. Thank you.

**Review by AbelHu — APPROVED** [review]

**Pull request merged**